StandardLibraryEssentials/Templates/TemplateBasics/task.md

```diff
@@ -0,0 +1,75 @@
+Imagine that you have a function for calculating the sum of two numbers. If you want to calculate the sum of two integers, you can write a function that takes two `int` numbers as arguments and returns their sum. However, suppose you also want to have such a function for `double`, `long long`, `float`, and a custom type such as `Fraction`. You will have to write a separate function for each type. This becomes inconvenient, as you end up writing multiple functions that essentially do the same thing but with different data types.
+
+Templates are designed to address this exact issue. They allow you to write a single function that can work with different types. For instance, you can write a template function that takes two arguments of any type and returns their sum as a variable of the same type.
+
+The syntax for creating a template function is simple: you start with the `template` keyword, followed by the `typename` keyword, and then include the type name you want to use as a template parameter. Then, you can use this type name in the function as if it were a regular type:
+
+```cpp
+template <typename T>
+T sum(T a, T b) {
+    return a + b;
+}
+```
+
+The number of template parameters is not limited to one. If you need to use multiple types for the function arguments, you can apply several template parameters. This is an example function featuring two:
+```cpp
+template <typename T1, typename T2>
+T1 sum(T1 a, T2 b) {
+    return a + b;
+}
+```
+This function will work with types such that the second type can be cast to the first type, and it supports the `+` operator. Note that the return type of the function is `T1`, so the result of the sum will be of the same type as the first argument.
+
+<div class="hint">
+    You can use `std::is_same` to check if two types are the same. This can be useful when you want to restrict the types that can be used with the template function.
+</div>
```

Comment:
We have not introduced type traits, it is a separate complicated topic, let's omit it here.